rust/sdk-processor/src/steps/events_processor/events_storer.rs

```diff
@@ -0,0 +1,102 @@
+use crate::{
+    config::db_config::PostgresConfig,
+    db::common::models::events_models::events::EventModel,
+    utils::database::{execute_in_chunks, get_config_table_chunk_size, ArcDbPool},
+};
+use ahash::AHashMap;
+use anyhow::Result;
+use aptos_indexer_processor_sdk::{
+    steps::{async_step::AsyncRunType, AsyncStep},
+    traits::{NamedStep, Processable},
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use async_trait::async_trait;
+use diesel::{
+    pg::{upsert::excluded, Pg},
+    query_builder::QueryFragment,
+    ExpressionMethods,
+};
+use processor::schema;
+use tracing::debug;
+pub struct EventsStorer
+where
+    Self: Sized + Send + 'static,
+{
+    conn_pool: ArcDbPool,
+    postgres_config: PostgresConfig,
+}
+
+impl EventsStorer {
+    pub fn new(conn_pool: ArcDbPool, postgres_config: PostgresConfig) -> Self {
+        Self {
+            conn_pool,
+            postgres_config,
+        }
+    }
+}
+
+fn insert_events_query(
+    items_to_insert: Vec<EventModel>,
+) -> (
+    impl QueryFragment<Pg> + diesel::query_builder::QueryId + Send,
+    Option<&'static str>,
+) {
+    use schema::events::dsl::*;
+    (
+        diesel::insert_into(schema::events::table)
+            .values(items_to_insert)
+            .on_conflict((transaction_version, event_index))
+            .do_update()
+            .set((
+                inserted_at.eq(excluded(inserted_at)),
+                indexed_type.eq(excluded(indexed_type)),
+            )),
+        None,
+    )
+}
+
+#[async_trait]
+impl Processable for EventsStorer {
+    type Input = EventModel;
+    type Output = EventModel;
+    type RunType = AsyncRunType;
+
+    async fn process(
+        &mut self,
+        events: TransactionContext<EventModel>,
+    ) -> Result<Option<TransactionContext<EventModel>>, ProcessorError> {
+        let per_table_chunk_sizes: AHashMap<String, usize> =
+            self.postgres_config.per_table_chunk_sizes.clone();
+        let execute_res = execute_in_chunks(
+            self.conn_pool.clone(),
+            insert_events_query,
+            &events.data,
+            get_config_table_chunk_size::<EventModel>("events", &per_table_chunk_sizes),
+        )
+        .await;
+        match execute_res {
+            Ok(_) => {
+                debug!(
+                    "Events versions {} to {} stored successfully",
```

Comment:
For logging, I generally recommend to log range as  `[start_version, end_version]` to make it more clear the end_version is inclusive.